### PR TITLE
Location setter and comparison fixes

### PR DESCRIPTION
**Pull request by padupr**
Setting a location now copies the location and location equality now also requires an equal path.

### CODE REVIEW

File: include/vara/Feature/Feature.h [hunk 1]
```diff
@@ -159,8 +159,8 @@ class Feature {
   [[nodiscard]] FeatureSourceRange *getFeatureSourceRange() {
     return Source.has_value() ? &Source.value() : nullptr;
   }
-  void setFeatureSourceRange(FeatureSourceRange FeatureSR) {
-    Source = std::move(FeatureSR);
+  void setFeatureSourceRange(FeatureSourceRange &FeatureSR) {
+    Source = FeatureSourceRange(FeatureSR);
```
Comment (vulder) [inline include/vara/Feature/Feature.h]:
```suggestion
  void setFeatureSourceRange(FeatureSourceRange FeatureSR) {
    Source = std::move(FeatureSR);
```
The original should actually be correct. The FeatureSR is copied into this function, therefore, we are allowed to move it. If I revert you change here,. all feature-unitttests and python tests pass on my system.

Could you please revert it locally and double check if the test fails afterwards.

Comment (padupr) [reply include/vara/Feature/Feature.h]: The original seems to work. I am not sure why it didn't work before. Maybe I forgot to update the pip package...

Comment (vulder) [reply include/vara/Feature/Feature.h]: Maybe, but at least we now have a test for it :)

**Review by vulder — APPROVED** [review]

File: bindings/python/vara-feature/pybind_FeatureSourceRange.cpp [hunk 2]
```diff
@@ -49,7 +49,12 @@ void init_feature_location_module(py::module &M) {
       .def_property_readonly(
           "end", &vf::FeatureSourceRange::getEnd,
           R"pbdoc(Get the end `LineColumnOffset` of this `Location`.)pbdoc")
-      .def("__str__", &vf::FeatureSourceRange::toString);
+      .def("__str__", &vf::FeatureSourceRange::toString)
+      .def("__eq__", [](const vf::FeatureSourceRange &Self,
+                        const vf::FeatureSourceRange *Other) {
+        if (!Other) return false;
```
Comment (vulder) [inline bindings/python/vara-feature/pybind_FeatureSourceRange.cpp]:
```suggestion
        if (!Other) {
            return false;
        }
```
please always use brackets, a if without is a bug ready to happen

File: bindings/python/vara-feature/pybind_FeatureSourceRange.cpp [hunk 3]
```diff
@@ -63,9 +68,9 @@ void init_feature_location_module(py::module &M) {
...
           &vf::FeatureSourceRange::FeatureSourceLocation::setColumnOffset)
       .def("__str__", &vf::FeatureSourceRange::FeatureSourceLocation::toString)
-      .def("__eq__",
-           [](const vf::FeatureSourceRange::FeatureSourceLocation &Self,
-              const vf::FeatureSourceRange::FeatureSourceLocation &Other) {
-             return Self == Other;
-           });
+      .def("__eq__", [](const vf::FeatureSourceRange::FeatureSourceLocation &Self,
+                        const vf::FeatureSourceRange::FeatureSourceLocation *Other) {
+        if (!Other) return false;
```
Comment (vulder) [inline bindings/python/vara-feature/pybind_FeatureSourceRange.cpp]:
```suggestion
        if (!Other) {
            return false
        };
```
same

File: bindings/python/tests/feature/test_feature.py [hunk 4]
```diff
@@ -136,6 +136,36 @@ def test_feature_location_setters(self):
...
+        test_feature_0.location = test_feature_0.location
+        self.assertTrue(test_feature_0.location)
+        self.assertEqual(test_feature_0.location.path, path)
+        self.assertEqual(test_feature_0.location.start.line_number,
+                         4)
+        self.assertEqual(
+            test_feature_0.location.start.column_offset, 2)
+        self.assertEqual(test_feature_0.location.end.line_number, 4)
+        self.assertEqual(test_feature_0.location.end.column_offset,
+                         18)
```
Comment (vulder) [inline bindings/python/tests/feature/test_feature.py]: This test code looks like the test case bellow. Does this make sense here?
If it does, this should be a different test case which specifically checks for self assignment for location setters.

**Review by vulder — CHANGES_REQUESTED** [review]

**Review by vulder — APPROVED** [review]

**Pull request merged**